starky/src/vars.rs

```diff
@@ -0,0 +1,26 @@
+use plonky2::field::field_types::Field;
+use plonky2::field::packed_field::PackedField;
+use plonky2::iop::ext_target::ExtensionTarget;
+
+#[derive(Debug, Copy, Clone)]
+pub struct StarkEvaluationVars<'a, F, P, const COLUMNS: usize, const PUBLIC_INPUTS: usize>
+where
+    F: Field,
+    P: PackedField<Scalar = F>,
+{
+    pub local_values: &'a [P; COLUMNS],
+    pub next_values: &'a [P; COLUMNS],
+    pub public_inputs: &'a [P::Scalar; PUBLIC_INPUTS],
```

Comment:
How will the public inputs work exactly? With this line it seems like we'll have `PUBLIC_INPUTS` columns of public inputs, is that right?

Reply:
I was thinking it could be up to the `Stark` impl. For example if using a STARK to do a hash-based signature, we'd want just the final row (i.e. the 4 outputs) to be public, so the eval methods would call

```rust
yield_constr.one_last_row(vars.local_values[i] - vars.public_inputs[i]);
```

for `i in [0 .. 3]`. Does that make sense?